src/sentry/tasks/summaries/weekly_reports.py

```diff
@@ -198,17 +137,76 @@ def prepare_organization_report(
     with sentry_sdk.start_span(op="weekly_reports.user_project_ownership"):
         user_project_ownership(ctx)
     with sentry_sdk.start_span(op="weekly_reports.project_event_counts_for_organization"):
-        project_event_counts_for_organization(ctx)
+        event_counts = project_event_counts_for_organization(
+            start=ctx.start,
+            end=ctx.end,
+            org_id=ctx.organization.id,
+            referrer="weekly_reports.outcomes",
+        )
+    for data in event_counts:
+        project_id = data["project_id"]
+        # Project no longer in organization, but events still exist
+        if project_id not in ctx.projects:
+            continue
+        project_ctx = ctx.projects[project_id]
+        total = data["total"]
+        timestamp = int(to_timestamp(parse_snuba_datetime(data["time"])))
+        if data["category"] == DataCategory.TRANSACTION:
+            # Transaction outcome
+            if data["outcome"] == Outcome.RATE_LIMITED or data["outcome"] == Outcome.FILTERED:
+                project_ctx.dropped_transaction_count += total
+            else:
+                project_ctx.accepted_transaction_count += total
+                project_ctx.transaction_count_by_day[timestamp] = total
+        elif data["category"] == DataCategory.REPLAY:
+            # Replay outcome
+            if data["outcome"] == Outcome.RATE_LIMITED or data["outcome"] == Outcome.FILTERED:
+                project_ctx.dropped_replay_count += total
+            else:
+                project_ctx.accepted_replay_count += total
+                project_ctx.replay_count_by_day[timestamp] = total
+        else:
+            # Error outcome
+            if data["outcome"] == Outcome.RATE_LIMITED or data["outcome"] == Outcome.FILTERED:
+                project_ctx.dropped_error_count += total
+            else:
+                project_ctx.accepted_error_count += total
+                project_ctx.error_count_by_day[timestamp] = (
+                    project_ctx.error_count_by_day.get(timestamp, 0) + total
+                )
 
     with sentry_sdk.start_span(op="weekly_reports.organization_project_issue_substatus_summaries"):
-        organization_project_issue_substatus_summaries(ctx)
+        substatus_counts = organization_project_issue_substatus_summaries(
```

Comment:
I refactored some of these functions to be passed stuff on `OrganizationReportContext` to be more reusable, but then I ended up using that class again 🙃 I can undo this.